File: 804-rotated-digits/rotated-digits.java
class Solution {
    public int rotatedDigits(int n) {
        int ans = 0 ;
        for(int i = 0 ; i<=n ; i++){    
            if(count(i)>0)ans++ ;
        }
        return ans ;
    }
    public static int count(int n){
        int count = 0 ;
        while(n>0){
            int r = n%10 ;
            if (r == 3 || r == 4 || r == 7) return 0;
            if (r == 2 || r == 5 || r == 6 || r == 9) count++;
            n/=10 ;
        }
        return count ;
    }
}

File: 804-rotated-digits/rotated-digits.cpp
class Solution {
public:
    int rotatedDigits(int n) {
        int ans = 0 ;
        for(int i = 0 ; i<=n ; i++){
            if(check(i))ans++ ;
        }
        return ans ;
    }
    bool check(int n){
        bool a = false ;
        while(n>0){
            int rem = n%10 ;
            if(rem==3 || rem==4 || rem==7)return false ;
            if(rem == 2 || rem == 5 || rem == 6 || rem == 9) a = true;
            n/=10 ;
        }
        return a ;
    }
};